lib/puppet/trusted_external.rb

```diff
@@ -3,11 +3,35 @@ module Puppet::TrustedExternal
   def retrieve(certname)
     command = Puppet[:trusted_external_command]
     return nil unless command
+    unless Puppet::FileSystem.directory?(command)
+      return fetch_data(command, certname)
+    end
+    # command is a directory. Thus, data is a hash of <basename> => <data> for
+    # each executable file in command. For example, if the files 'servicenow.rb',
+    # 'unicorn.sh' are in command, then data is the following hash:
+    #   { 'servicenow' => <servicenow.rb output>, 'unicorn' => <unicorn.sh output> }
+    data = {}
+    Puppet::FileSystem.children(command).each do |file|
+      executable_file = Puppet::FileSystem.file?(file) && Puppet::FileSystem.executable?(file)
+      next unless executable_file
```

Comment:
I can see someone forgetting to set the executable bit and wondering why it's not working. Maybe warn or issue a debug that the file is being skipped? What does facter do when there's an external fact that isn't executable?

Reply:
Looks like Facter just prints out something like

```
2020-05-26 18:26:05.401919 DEBUG puppetlabs.facter - skipping file "/root/EXTERNAL/script.sh": No resolver for external facts file /root/EXTERNAL/script.sh
```

I agree that at least a debug message would be useful so went ahead and added some